### PR TITLE
feat(templates): add templates package

### LINKED ISSUE #10109: Create new package `patternfly-widgets` in `patternfly-react`

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #10109 and #10111

Adds the new package to repo with basic template, and sets up documentation.

~~https://github.com/patternfly/patternfly-org/pull/3902 needs to be merged before docs build will succeed. But first I want to confirm the starting version of `1.0.0-alpha.0`. Are we good starting from `1` or do we want to start from `5` to match other PF packages?~~

~~What does everyone think about `react-templates` as the package name as well?~~

### CODE REVIEW

File: packages/react-templates/src/components/Select/SimpleSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        ref={innerRef}
+        {...props}
+      >
+        <SelectList>{initialOptions}</SelectList>
+      </Select>
+    </React.Fragment>
+  );
+};
+
+export const SimpleSelect = React.forwardRef((props: SimpleSelectProps, ref: React.Ref<any>) => (
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/SimpleSelect.tsx:78]: Can we actually call this `SelectSimple` so the related componet is named first?

File: packages/react-templates/src/components/Select/examples/SelectTemplates.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+id: Select
+section: components
+subsection: menus
+template: true
+propComponents: ['SimpleSelect']
+---
+
+Note: Templates live in their own package at [@patternfly/react-templates](https://www.npmjs.com/package/@patternfly/react-templates)!
+
+For custom use cases, please the select component suite from [@patternfly/react-core](https://www.npmjs.com/package/@patternfly/react-core).
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/examples/SelectTemplates.md:11]:
```suggestion
For custom use cases, please see the select component suite from [@patternfly/react-core](https://www.npmjs.com/package/@patternfly/react-core).
```

File: packages/react-templates/src/components/Select/examples/SimpleSelectDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/examples/SimpleSelectDemo.tsx:1]: I would also suggest naming this `SelectSimpleDemo`.

File: packages/react-templates/src/components/Select/examples/SelectTemplates.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+id: Select
+section: components
+subsection: menus
+template: true
+propComponents: ['SimpleSelect']
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/examples/SelectTemplates.md:6]: The prop table is not generated.  Does the org PR have to go on first?

File: packages/react-templates/src/components/Select/SimpleSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        onSelect={_onSelect}
+        onOpenChange={(isOpen) => setIsOpen(isOpen)}
+        toggle={toggle}
+        shouldFocusToggleOnSelect
+        ref={innerRef}
+        {...props}
+      >
+        <SelectList>{initialOptions}</SelectList>
+      </Select>
+    </React.Fragment>
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/SimpleSelect.tsx:74]: I don't think you need the fragment here.

File: packages/react-templates/src/components/Select/SimpleSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  );
+};
+
+export const SimpleSelect = React.forwardRef((props: SimpleSelectProps, ref: React.Ref<any>) => (
+  <SimpleSelectBase {...props} innerRef={ref} />
+));
+
+SimpleSelect.displayName = 'SelectOption';
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/SimpleSelect.tsx:82]: Should the displayName match the componet name?

**Review by tlabaj — CHANGES_REQUESTED** [review]
Looking good!  I would also make the update to the Select componet to add a blurb referring consumers to the templates.  Or does that blurb go in Org?

File: packages/react-templates/src/components/Select/examples/SelectTemplates.md (hunk 4 above)
Comment (kmcfaul) [reply packages/react-templates/src/components/Select/examples/SelectTemplates.md:6]: Yeah I think we have to update the MD routes in org as well for that. cc @nicolethoen

**Comment by kmcfaul** [comment]
As part of updating the docs-framework, this PR includes a fix for `react-code-editor` which adds `monaco-editor` as a dependency (which was removed in the docs-framework).

File: packages/react-templates/src/components/Select/SelectSimple.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onToggleClick}
+      isExpanded={isOpen}
+      isDisabled={isDisabled}
+      style={
+        {
+          width: '200px'
+        } as React.CSSProperties
+      }
```
Comment (thatblindgeye) [inline packages/react-templates/src/components/Select/SelectSimple.tsx:53]: Not a blocker for this, but wondering if we want to hardcode this. The content truncates fine, but I imagine consumers may want to control the width to some degree.

File: packages/react-templates/src/components/Select/SelectSimple.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Select, SelectList } from '@patternfly/react-core/dist/esm/components/Select';
+import { MenuToggle, MenuToggleElement } from '@patternfly/react-core/dist/esm/components/MenuToggle';
+
+export interface SelectSimpleProps {
+  /** @hide Forwarded ref */
+  innerRef?: React.Ref<any>;
+  /** Initial options of the select. */
+  initialOptions?: React.ReactElement[];
```
Comment (thatblindgeye) [inline packages/react-templates/src/components/Select/SelectSimple.tsx:9]: Probably want to decide what exactly is passed to this prop, either an array of SelectOptions or an array of objects like in the [checkbox template PR](https://github.com/patternfly/patternfly-react/pull/10159/commits/27a8be04ff005b194773305800b2f8779b7fb6fb#diff-f739789cccb17b7e51546f4e5b4e8c9c470a047ade976321a276e47ca3e8ddaa). Keeping the type consistent would probably be less confusing for consumers if they're using both templates.

**Review by thatblindgeye — COMMENTED** [review]
This is looking solid 🔥 Just had some comments below

File: packages/react-templates/src/components/Select/SelectSimple.tsx (hunk 8 above)
Comment (wise-king-sullyman) [reply packages/react-templates/src/components/Select/SelectSimple.tsx:9]: IMO I think we should just launch all of these in beta and come back and make the API's consistent once we know where we land on all of our planned templates.

**Review by wise-king-sullyman — APPROVED** [review]
Other than Eric's comments (and a request that we add tests, that can be a followup though) LGTM!

File: packages/react-templates/src/components/Select/SelectSimple.tsx (hunk 7 above)
Comment (kmcfaul) [reply packages/react-templates/src/components/Select/SelectSimple.tsx]: Added a prop

File: packages/react-templates/src/components/Select/SelectSimple.tsx (hunk 8 above)
Comment (kmcfaul) [reply packages/react-templates/src/components/Select/SelectSimple.tsx]: Updated to SelectOptionProps[] but we can revisit props after other templates have gone in to review consistency.

**Review by thatblindgeye — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]
LGTM!

**Pull request merged**